include/boost/geometry/arithmetic/general_form.hpp

```diff
@@ -0,0 +1,161 @@
+namespace arithmetic
+{
+
+//--------------------------------------------------------------------------
+// Structure containing the General Form of a line, a*x + b*y + c == 0
+// Might be conceptized later. Therefore operations are implemented outside
+// the structure itself.
+template <typename GeneralType = double>
+struct general_form
+{
+    general_form()
+        : a(GeneralType())
+        , b(GeneralType())
+        , c(GeneralType())
+        , normalized(false)
+    {}
+
+    // Horizontal: a == 0, for example y-3=0, y==3
+    // Vertical: b == 0, for example x-2=0, x==2
+    // Through origin: c == 0
+    GeneralType a;
+    GeneralType b;
+    GeneralType c;
+    bool normalized;
+};
+
+template <typename GeneralType, typename Coordinate>
+inline
+general_form<GeneralType> construct_line(Coordinate const& x1,
+    Coordinate const& y1, Coordinate const& x2, Coordinate const& y2)
+{
+    general_form<GeneralType> result;
+    result.a = y1 - y2;
+    result.b = x2 - x1;
+    result.c = -result.a * x1 - result.b * y1;
+    return result;
+}
+
+template <typename T, typename Point>
+inline general_form<T> construct_line(Point const& a, Point const& b)
+{
+    return construct_line<T>(geometry::get<0>(a),
+                             geometry::get<1>(a),
+                             geometry::get<0>(b),
+                             geometry::get<1>(b));
+}
+
+template <typename T, typename Segment>
+inline general_form<T> construct_line(Segment const& segment)
+{
+    return construct_line<T>(geometry::get<0, 0>(segment),
+                             geometry::get<0, 1>(segment),
+                             geometry::get<1, 0>(segment),
+                             geometry::get<1, 1>(segment));
+}
+
+
+// Calculates intersection point of two infinite lines.
+// Returns true if the lines intersect.
+// Returns false if lines are parallel (or collinear, possibly opposite)
+template <typename Point, typename GeneralType>
+inline
+bool get_intersection(Point& ip,
+                      general_form<GeneralType> const& p,
+                      general_form<GeneralType> const& q)
+{
+    GeneralType const denominator = p.b * q.a - p.a * q.b;
+
+    static GeneralType const zero = 0;
+    if (math::equals(denominator, zero))
+    {
+        // Lines are parallel
+        return false;
+    }
+
+    // Calculate the intersection coordinates
+    geometry::set<0>(ip, (p.c * q.b - p.b * q.c) / denominator);
+    geometry::set<1>(ip, (p.a * q.c - p.c * q.a) / denominator);
+
+    return true;
+}
+
+//! Return a distance-side-measure for a point to a line
+//! Point is located left of the line if value is positive,
+//! right of the line is value is negative, and on the line if the value
+//! is exactly zero
+template <typename GeneralType, typename CoordinateType>
+inline
+typename select_most_precise<GeneralType, CoordinateType>::type
+side_value(general_form<GeneralType> const& f,
+    CoordinateType const& x, CoordinateType const& y)
```

Comment:
Right, so this is another 2d-only function, but here it's ok since it takes `general_form` so it would be possible to implement n-d version by overloading it.